src/main/java/ch/wisv/areafiftylan/controller/EventRestController.java

```diff
@@ -0,0 +1,156 @@
+        this.eventService = eventService;
+        this.teamService = teamService;
+    }
+
+    /**
+     * Add a new Event
+     *
+     * @param eventDTO EventDTO containing all info about the Event. Validated for null fields
+     *
+     * @return Message containing the location of the created Event
+     */
+    @RequestMapping(method = RequestMethod.POST)
+    @PreAuthorize("hasRole('ADMIN')")
+    public ResponseEntity<?> addEvent(@Validated @RequestBody EventDTO eventDTO) {
+        Event event = eventService.addEvent(eventDTO);
+
+        HttpHeaders httpHeaders = new HttpHeaders();
+        httpHeaders.setLocation(
+                ServletUriComponentsBuilder.fromCurrentRequest().path("/{id}").buildAndExpand(event.getId()).toUri());
+
+        return createResponseEntity(HttpStatus.CREATED, httpHeaders,
+                "Event successfully created at " + httpHeaders.getLocation(), event);
+    }
+
+    /**
+     * Get all events
+     *
+     * @return A collection of all current events
+     */
+    @JsonView(View.Public.class)
+    @RequestMapping(method = RequestMethod.GET)
+    public Collection<Event> getAllEvents() {
+        return eventService.getAllEvents();
+    }
+
+    /**
+     * Update the event with the given eventId
+     *
+     * @param eventDTO DTO containing the fields that need to be updated. Can contain null fields
+     * @param eventId  Id of the event to be updated
+     *
+     * @return Message containing the updated event
+     */
+    @PreAuthorize("hasRole('ADMIN')")
+    @RequestMapping(value = "/{eventId}", method = RequestMethod.PUT)
+    public ResponseEntity<?> updateEvent(@RequestBody EventDTO eventDTO, @PathVariable Long eventId) {
+        Event event = eventService.updateEvent(eventId, eventDTO);
+        return createResponseEntity(HttpStatus.OK, "Event successfully updated", event);
+    }
+
+    /**
+     * Delete the event with the given ID
+     *
+     * @param eventId ID of the event to be deleted
+     *
+     * @return The deleted event
+     */
+    @PreAuthorize("hasRole('ADMIN')")
+    @RequestMapping(value = "/{eventId}", method = RequestMethod.DELETE)
+    public ResponseEntity<?> updateEvent(@PathVariable Long eventId) {
+        Event event = eventService.deleteEvent(eventId);
+        return createResponseEntity(HttpStatus.OK, "Event successfully deleted", event);
+    }
+
+    /**
+     * Register a team for an event. Only for captains and admin
+     *
+     * @param teamId  Id of the Team to be registered
+     * @param eventId Id of the Event to be registered for.
+     *
+     * @return Message containing the result of the registration
+     */
+    @PreAuthorize("@currentUserServiceImpl.canEditTeam(principal, #teamId)")
+    @RequestMapping(value = "/{eventId}", method = RequestMethod.POST)
+    public ResponseEntity<?> registerTeamForEvent(@RequestBody Long teamId, @PathVariable Long eventId) {
+        Team team = teamService.getTeamById(teamId);
```

Comment:
I think the teamId should be passed on to the registerTeam method. Or the Event should be retrieved using the eventId. I prefer the first one.